src/api/error.rs

```diff
@@ -0,0 +1,74 @@
+// LNP/BP Rust Library
+// Written in 2020 by
+//     Dr. Maxim Orlovsky <orlovsky@pandoracore.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the MIT License
+// along with this software.
+// If not, see <https://opensource.org/licenses/MIT>.
+
+
+use zmq;
+
+use bitcoin;
+use bitcoin::secp256k1;
+
+#[cfg(feature="use-rgb")]
+use crate::csv;
+
+
+#[derive(Debug, Display)]
+#[display_from(Debug)]
+pub enum Error {
```

Comment:
I expect the API to be in flux for some time, so making the enum private and bundling it into a newtype struct might improve forward compatibility for the time being.

Reply:
An advantage of struct over `non_exhaustive` is that it can change its internal representation without breaking backwards compatibility. This may be important in the early stage. Once possible errors are well-understood, making the enum public (possibly with `non_exhaustive`) will be fine.